cdp/smart_contract.py

```diff
@@ -0,0 +1,338 @@
+import json
+import time
+from enum import Enum
+from typing import Any
+
+from eth_account.signers.local import LocalAccount
+
+from cdp.cdp import Cdp
+from cdp.client.models.create_smart_contract_request import CreateSmartContractRequest
+from cdp.client.models.deploy_smart_contract_request import DeploySmartContractRequest
+from cdp.client.models.multi_token_contract_options import MultiTokenContractOptions
+from cdp.client.models.nft_contract_options import NFTContractOptions
+from cdp.client.models.smart_contract import SmartContract as SmartContractModel
+from cdp.client.models.smart_contract_options import SmartContractOptions
+from cdp.client.models.token_contract_options import TokenContractOptions
+from cdp.transaction import Transaction
+
+
+class SmartContract:
+    """A representation of a SmartContract on the blockchain."""
+
+    class Type(Enum):
+        """Enumeration of SmartContract types."""
+
+        ERC20 = "erc20"
+        ERC721 = "erc721"
+        ERC1155 = "erc1155"
+
+        def __str__(self) -> str:
+            """Return a string representation of the Type."""
+            return self.value
+
+        def __repr__(self) -> str:
+            """Return a string representation of the Type."""
+            return str(self)
+
+    class TokenContractOptions(dict[str, Any]):
```

Comment:
We cannot inherit from `dict[str, Any]` because it is a type annotation not a class. 
```suggestion
    class TokenContractOptions(dict):
```